packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts

```diff
@@ -176,10 +177,22 @@ export class ChainAdapter
         to,
         wallet,
         bip44Params = ChainAdapter.defaultBIP44Params,
-        chainSpecific: { satoshiPerByte, accountType },
+        chainSpecific: { satoshiPerByte, accountType, opReturnData },
         sendMax = false
       } = tx
 
+      // coinselect doesnt work with op_return outputs
+      // This hack adjusts the fee up proportionally to the op_return length
+      // op_return is ultimately added to the transaction upon signing but not here
+      // Imperfect because the assumption of 300 bytes for a standard tx could vary with different inputs/outputs
+      // Another solution would be to write our own coinselect algorithm
+      const standardTxBytes = 300
+      const opReturnTxBytes = standardTxBytes + (opReturnData?.length ?? 0)
+      const adjustedSatsPerByte = bn(satoshiPerByte)
+        .times(bn(opReturnTxBytes).div(standardTxBytes))
+        .dp(0)
+        .toString()
+
```

Comment:
it seems like we should do all of this within bn, and we're making a string to immediately convert to a number into coinselect?